flask_app/issue_trackers.py

```diff
@@ -38,17 +39,23 @@ def __init__(self, *, url, config):
         self._jira = JIRAAPI(url, basic_auth=(config['username'], config['password']))
         self._resolution_grace = timedelta(days=config.get('resolution_grace', 0))
 
+    def _refresh_issue(self, issue_obj):
+        issue = self._jira.issue(issue_obj.id_in_tracker)
+        if issue.fields.resolutiondate is None:
+            issue_obj.open = True
+        else:
+            resolution_date = flux.current_timeline.datetime.strptime(
+                issue.fields.resolutiondate, '%Y-%m-%dT%H:%M:%S.%f%z')
+            now = flux.current_timeline.datetime.now().replace(tzinfo=timezone.utc)
+            issue_obj.open = (now - resolution_date) < self._resolution_grace
+
+    @needs_app_context
     def refresh(self, issues):
         for issue_obj in issues:
-            issue = self._jira.issue(issue_obj.id_in_tracker)
-
-            if issue.fields.resolutiondate is None:
-                issue_obj.open = True
-            else:
-                resolution_date = flux.current_timeline.datetime.strptime(
-                    issue.fields.resolutiondate, '%Y-%m-%dT%H:%M:%S.%f%z')
-                now = flux.current_timeline.datetime.now().replace(tzinfo=timezone.utc)
-                issue_obj.open = (now - resolution_date) < self._resolution_grace
+            try:
+                self._refresh_issue(issue_obj)
```

Comment:
You should call `db.session.commit()` after every successful call and `db.session.rollback()` on each exception.

I know that it wasn't there before but this is my fault.

When you update the first issue and don't call `db.session.commit()` a transaction is opened and the issue row is locked. Since this operation takes long time it means that you lock your rows for long time. Calling `db.session.commit()` on each success means that you release the lock from lines which their JIRA was updated.